### PR TITLE
Add extra wait_quit_zypper after s390x boot up

### PR DESCRIPTION
**Pull request by coolgw**
Add extra wait_quit_zypper after s390x boot up

- Related ticket: https://progress.opensuse.org/issues/104439
- Needles: na
- Verification run: https://openqa.suse.de/tests/7926824
NOTE: can not easy reproduce this issue with my fix after around 20 runs, current VR only show no side effect for exist test.

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -1407,7 +1407,19 @@ sub reconnect_mgmt_console {
...
-                    zypper_call("in susefirewall2-to-firewalld");
+                    my $ret;
+                    my $max_num = 3;
+                    # Sometimes wait_quit_zypper still not wait enough timeout
+                    for (my $i = 1; $i < $max_num + 1; $i = $i + 1) {
+                        $ret = zypper_call("in susefirewall2-to-firewalld", exitcode => [0, 7]);
+                        if ($ret == 7) {
+                            record_info('The ZYPP library is still locked, wait more seconds');
+                            wait_quit_zypper;
+                            next;
```
Comment (hjluo) [inline lib/utils.pm:1418]: No need to next here.

Comment (coolgw) [reply lib/utils.pm]: thansk! fixed.

**Comment by rfan1** [comment]
sed -i s/s390/s390x/g

File: lib/utils.pm [hunk 2]
```diff
@@ -1349,6 +1349,33 @@ sub _handle_login_not_found {
...
+ _handle_firewall();
+
+Internal helper function used by C<reconnect_mgmt_console>.
+
+=cut
+sub _handle_firewall {
+    select_console 'root-console';
+    if (script_run("iptables -S | grep 'A input_ext.*tcp.*dport 59.*-j ACCEPT'", 30) != 0) {
+        wait_quit_zypper;
+        my $ret;
```
Comment (rfan1) [inline lib/utils.pm:1363]: Is it a bug on s390x?
why locked there if no active daemon?

Comment (coolgw) [reply lib/utils.pm:1363]: not bug, this is caused by firewall upgrade

**Review by lemon-suse — APPROVED** [review]

File: lib/utils.pm [hunk 3]
```diff
@@ -1349,6 +1349,33 @@ sub _handle_login_not_found {
...
+            $ret = zypper_call("in susefirewall2-to-firewalld", exitcode => [0, 7]);
+            if ($ret == 7) {
+                record_info('The ZYPP library is still locked, wait more seconds');
+                wait_quit_zypper;
+            } else {
+                last;
+            }
+        }
+        susefirewall2_to_firewalld();
+    }
```
Comment (okurz) [inline lib/utils.pm:1376]:
```suggestion
    return if script_run("iptables -S | grep 'A input_ext.*tcp.*dport 59.*-j ACCEPT'", 30) == 0;
    wait_quit_zypper
    my $ret;
    my $max_num = 3;
    # Sometimes wait_quit_zypper still not wait enough timeout
    for (1 .. $max_num) {
        $ret = zypper_call("in susefirewall2-to-firewalld", exitcode => [0, 7]);
        if ($ret == 7) {
            record_info('The ZYPP library is still locked, wait more seconds');
            wait_quit_zypper;
        } else {
            last;
        }
    }
    susefirewall2_to_firewalld();
```

**Review by rfan1 — APPROVED** [review]

**Comment by rfan1** [comment]
One more comment,  when return code is 7:

           The ZYPP library is locked, e.g. packagekit is running.

Can we also add it into our consideration?

**Comment by coolgw** [comment]
> One more comment, when return code is 7:
> 
> ```
>        The ZYPP library is locked, e.g. packagekit is running.
> ```
> 
> Can we also add it into our consideration?

Base our failed case log this is not caused by packagekit.

**Pull request merged**